### DISCUSSION TITLE
I get an error: cannot attach kprobe

### DISCUSSION THREAD

**Discussion by xing393939** [bpftrace/bpftrace#2614] [Q&A]
when i exec the command `bpftrace -e 'kprobe:trace_ignore_this_task{}'`, it shows
```
cannot attach kprobe, Invalid argument
ERROR: Error attaching probe: 'kprobe:trace_ignore_this_task'
```

[this article](https://stackoverflow.com/questions/67766320/which-kprobe-hooks-can-i-attach-ebpf-programs-to) says `You can attach a kprobe to nearly all functions of your kernel`

why `trace_ignore_this_task` can not be attach? here is my env:
```
# grep trace_ignore_this_task /proc/kallsyms
ffffffff8ce1e580 T trace_ignore_this_task
# uname -r
5.19.0-41-generic
# bpftrace -V
bpftrace v0.14.0
```

**Comment by danobi** [comment]
Hi, this is a good question. The reason why bpftrace (and ftrace as well) cannot attach to that function is it’s not listed in /sys/kernel/debug/tracing/available_filter_functions. 

I’m still trying to figure out why the kernel won’t list it as traceable but I am assuming it’s to avoid recursion issues. I’m not quite finding where in the kernel the exemption occurs though. 

Hope that explains it. 

On Sun, May 21, 2023, at 6:23 AM, xing393939 wrote:
> when i exec the command `bpftrace -e 
> 'kprobe:trace_ignore_this_task{}'`, it shows
>
> `cannot attach kprobe, Invalid argument
> ERROR: Error attaching probe: 'kprobe:trace_ignore_this_task'
> ` 
> this article 
> <https://stackoverflow.com/questions/67766320/which-kprobe-hooks-can-i-attach-ebpf-programs-to> 
> says `You can attach a kprobe to nearly all functions of your kernel`
>
> why `trace_ignore_this_task` can not be attach? here is my env:
>
> `# grep trace_ignore_this_task /proc/kallsyms
> ffffffff8ce1e580 T trace_ignore_this_task
> # uname -r
> 5.19.0-41-generic
> # bpftrace -V
> bpftrace v0.14.0
> ` 
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/iovisor/bpftrace/discussions/2614>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAGJKJ6TDNKUQVQXILBU7BDXHHUINANCNFSM6AAAAAAYJJ2S6I>.
> You are receiving this because you are subscribed to this 
> thread.Message ID: ***@***.***>

-- 
  Daniel Xu
  ***@***.***

**Reply by danobi** [reply]
I believe this is why: https://github.com/torvalds/linux/blob/4927cb98f0eeaa5dbeac882e8372f4b16dc62624/kernel/trace/Makefile#L3-L5

The mcount stuff is disabled from the trace subsystem
